analysers/analyser_osmosis_indoor.py

```diff
@@ -0,0 +1,204 @@
+FROM
+    ways
+WHERE
+    is_polygon AND
+    tags != ''::hstore AND
+    tags?'indoor' AND
+    tags->'indoor' in ('room', 'corridor', 'area', 'level')
+"""
+
+sql01 = """
+CREATE INDEX indoor_surfaces_idx_geom on indoor_surfaces USING gist(geom)
+"""
+
+sql10 = """
+SELECT
+    id,
+    ST_AsText(way_locate(linestring))
+FROM
+    ways
+WHERE
+    tags != ''::hstore AND
+    NOT is_polygon AND
+    tags?'indoor' AND
+    tags->'indoor' in ('room', 'corridor', 'area', 'level')
+"""
+
+sql20 = """
+SELECT
+    id,
+    ST_AsText(geom)
+FROM
+    nodes
+WHERE
+    tags != ''::hstore AND
+    tags?'indoor' AND
+    tags->'indoor' in ('room', 'corridor', 'area', 'level')
+"""
+
+sql30 = """
+SELECT
+    public_indoor_rooms.id,
+    ST_AsText(way_locate(public_indoor_rooms.geom))
+FROM
+    indoor_surfaces AS public_indoor_rooms
+    LEFT JOIN nodes ON
+        public_indoor_rooms.geom && nodes.geom AND
+        nodes.id = ANY(public_indoor_rooms.nodes) AND
+        nodes.tags != ''::hstore AND
+        nodes.tags?'door'
+WHERE
+    public_indoor_rooms.indoor = 'room' AND
+    public_indoor_rooms.public_access AND
+    nodes.id IS NULL
+"""
+
+sql40 = """
+SELECT
+    id,
+    ST_AsText(way_locate(geom))
+FROM
+    indoor_surfaces
+WHERE
+    NOT tags?'level'
+"""
+
+sql50 = """
+SELECT
+    id,
+    ST_AsText(way_locate(geom))
+FROM
+    indoor_surfaces
+WHERE
+    indoor_surfaces.indoor != 'room' AND
+    indoor_surfaces.tags?'shop'
+"""
+
+sql60 = """
+CREATE TEMP TABLE indoor_surfaces_connected_to_highways AS
+SELECT DISTINCT
+    indoor_surfaces.id,
+    indoor_surfaces.geom,
+    indoor_surfaces.indoor,
+    indoor_surfaces.level AS surface_level,
+    highway_ends.nid,
+    highway_ends.highway,
+    highways.tags->'level' AS connected_highway_level
+FROM
+    indoor_surfaces
+    JOIN nodes ON
+        nodes.geom && indoor_surfaces.geom AND
+        nodes.id = ANY(indoor_surfaces.nodes)
+    JOIN highway_ends ON
+        highway_ends.nid = nodes.id
+    JOIN highways ON
+        highways.id = highway_ends.id
+WHERE
+    indoor_surfaces.indoor IN ('room', 'corridor', 'area') AND
+    highways.highway IN ('steps', 'footway', 'pedestrian')
+"""
+
+# TODO : check that all surfaces have at least one connected_highway_level that matches it own surface_level
+# (dealing with level="0;1" that occurs on highway=steps
+# and assuming than no level on highway is probably implicit level=0)
+
+sql61 = """
+SELECT
+    indoor_surfaces.id,
+    ST_AsText(way_locate(indoor_surfaces.geom))
+FROM
+    indoor_surfaces
+    LEFT JOIN indoor_surfaces AS indoor_surfaces_other ON
+        indoor_surfaces_other.id != indoor_surfaces.id AND
+        indoor_surfaces_other.geom && indoor_surfaces.geom AND
+        indoor_surfaces_other.nodes && indoor_surfaces.nodes
+    LEFT JOIN indoor_surfaces_connected_to_highways ON
+        indoor_surfaces_connected_to_highways.id = indoor_surfaces.id
+WHERE
+    indoor_surfaces.indoor IN ('room', 'corridor', 'area') AND
```

Comment:
We need to do check the same thing on `indoor_surfaces_other`, doesn't we?

Reply:
indeed, good point 👍
it's done